sdk/daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/preprocessing/Preprocessor.scala

```diff
@@ -219,11 +220,25 @@ private[engine] final class Preprocessor(
     }
 
   private[engine] def preprocessReplayCommand(
-      cmd: command.ReplayCommand
-  ): Result[speedy.Command] =
-    safelyRun(pullPackage(List(cmd.templateId))) {
+      cmd: ReplayCommand
+  ): Result[speedy.Command] = {
+    def templateAndInterfaceIds =
+      cmd match {
+        case ReplayCommand.Create(templateId, _) => List(templateId)
+        // Exercise includes interface package in main-3.x, to fix a bug with retroactive instances
+        // These aren't supported in 3x. Unclear if Exeercise should include the id.
+        // TODO: Clarify
```

Comment:
Note here, as https://github.com/digital-asset/daml/pull/19822  wasn't forward ported
Note also that it implements a test that we can no longer crib from. Once we have a decision on this PR, we should port over the test.

Reply:
I'm not sure I fully understand this comment:
>Note here, as #19822 wasn't forward ported

It seems to me this is (partly at least) a port of #19822

> Note also that it implements a test that we can no longer crib from. Once we have a decision on this PR, we should port over the test.

What kind of decision should we reach?